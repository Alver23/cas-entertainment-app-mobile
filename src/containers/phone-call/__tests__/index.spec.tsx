// Dependencies
import React from 'react';
import call from 'react-native-phone-call';
import { Button } from 'react-native-elements';
import { shallow, ShallowWrapper } from 'enzyme';

// Under test file
import PhoneCall from '../index';

// Mocks
jest.mock('react-native-phone-call', () => jest.fn());

describe('<PhoneCall />', () => {
  let component: ShallowWrapper;

  const props = {
    phoneNumber: 1234,
  };

  beforeEach(() => {
    component = shallow(<PhoneCall {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onPress void', () => {
    const onPress = component.find(Button).prop('onPress') as any;

    onPress();
    expect(call).toHaveBeenCalled();
  });
});
