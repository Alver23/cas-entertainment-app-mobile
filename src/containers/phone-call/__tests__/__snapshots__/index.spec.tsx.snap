// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneCall /> should save a snapshot of the component 1`] = `
<Fragment>
  <ForwardRef(Themed.Button)
    buttonStyle={
      Object {
        "marginVertical": 24,
      }
    }
    containerStyle={
      Object {
        "width": "40%",
      }
    }
    icon={
      Object {
        "name": "phone",
        "size": 20,
        "type": "feather",
      }
    }
    onPress={[Function]}
    title="Llamar"
    titleStyle={
      Object {
        "fontSize": 26.5,
      }
    }
  />
</Fragment>
`;
